### PR TITLE
Readme and setup

### PR DESCRIPTION
**Pull request by j042**
# Overview
Readme extension

### CODE REVIEW

**Comment by j042** [comment]
@jzwar can you please fill out `compose()`, `compose_derivative()`, and multi patch?

File: README.md [hunk 1]
```diff
@@ -1,37 +1,168 @@
 # splinepy
 [![workflow](https://github.com/tataratat/splinepy/actions/workflows/main.yml/badge.svg)](https://github.com/tataratat/splinepy/actions)
```
Comment (jzwar) [inline README.md:2]: Can this line be deleted? I consider this as internal

File: README.md [hunk 2]
```diff
@@ -1,37 +1,168 @@
...
+        [0.5, 0. ],
+        [1. , 0. ],
+        [0. , 1. ],
+        [0.5, 1. ],
+        [1. , 1. ],
+    ],
+)
+
+# We always store control points in 2D arrays with
+# (total_control_points, physical_dimension) shape.
```
Comment (jzwar) [inline README.md:60]:
```suggestion
# (total_number_of_control_points, physical_dimension) shape.
```

File: README.md [hunk 3]
```diff
@@ -1,37 +1,168 @@
...
+# They fill control grid by iterating lower-indexed dimensions first.
+# But if you are rather familiar with grid-like structure, this should hold
```
Comment (jzwar) [inline README.md:62]:
```suggestion
# But if you prefer the grid-like structure, this should hold
```

File: README.md [hunk 4]
```diff
@@ -1,37 +1,168 @@
...
+ - boundary assignement using different techniques, relying either on the boundary position or on the continuity inbetween patches
+ - Boundary extraction
+
+### IO
+Available in `splinepy.io`.
+| Formats | Description |
+| ------- | ----------- |
+| iges | Loads/Exports splines from an [IGES](https://en.wikipedia.org/wiki/IGES) file |
+| irit | [IRIT](https://www.cs.technion.ac.il/~irit/) compatible format |
+| json | Supports base64 encoding |
```
Comment (jzwar) [inline README.md:143]:
```suggestion
| json | (Custom) easy-to-read format, supports base64 encoding |
```

**Review by jzwar — APPROVED** [review]
Very nice, finally an overview on all the features. Nice to see how far this project has come

File: README.md (hunk 1 above)
Comment (j042) [reply README.md:2]: this is deploy and full build status for, thought it's standard

Comment (jzwar) [reply README.md:2]: its just a hyperlink. there is no nice image

Comment (j042) [reply README.md:2]: there was a mismatch in naming. it works now and now I am gonna rename the workflow real quick

**Pull request merged**